### ISSUE TITLE
VaAccordion expand/collapse all accordion items not 508 friendly

### ISSUE THREAD

**Issue by imelda-guzman** [department-of-veterans-affairs/vets-design-system-documentation#1663]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

When using a screen reader such as JAWS and clicking the "Expand all" accordion button JAWS does not read back to a user that all the accordion items are expanded.

## What I expected to happen

After clicking the "Expand all" button, I expected to be alerted that all accordion items are now expanded and vice versa when "Collapse all" button is clicked.

I have tried to create my own alert state when expand/collapse button is clicked but this does not work as the button has no event listener I can tie into.  It appears the event listener is only available on the individual accordion items.

## Reproducing
- Component Library version: 6.2.3
- Web Components version: 2.14.0
- Formation version: 7.0.1
- Device: HP laptop
- Browser: Microsoft Edge

Steps to reproduce:

1. Setup and accordion with 3 items.
2. Using a screen reader such as JAWS, navigate to the "Expand all" button and listen to instructions for expanding all accordions.
3. Click on "Expand all" button and notice the screen reader says nothing to notify user all accordion items are now expanded.
4. Click "Collapse all" button and notice the screen reader says nothing to notify user all accordion items are now collapsed.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [X] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details
I am requesting an improvement to this component as it currently does not meet 508 compliance, or at the very least add an event to the expand/collapse all button so I may be able to write a custom alert state for screen readers.

**Comment by caw310** [comment]
Adding to sprint 24 since this is a blocker and DST engineer has time to look into this.

**Comment by powellkerry** [comment]
I have done some research and both chromeVox and mac voice over do not read a result of clicking the expand-all/collapse-all button. Additionally I have found 1 solution to this, but am not sure if this is the correct way to go about adding this functionality:

Add an aria-live div to convey the result (va-accordion.tsx:217):
```
<div aria-live="polite" aria-atomic="true" class="sr-button-result" ref={el => (this.srButtonResult = el as HTMLDivElement)}></div>
```

Add TypeScript type for the reference (va-accordion.tsx:57):
```
srButtonResult!: HTMLDivElement;
```

Update expandCollapseAll function to set the text (va-accordion.tsx:144) *This will require a new i18n translated string
```
if (expanded) {
      this.srButtonResult.innerText="All Expanded";
    } else {
      this.srButtonResult.innerText="All Collapsed";
    }
    setTimeout(() => {
      this.srButtonResult.innerText = "";
    }, 5000)
```
Add CSS to hide the div for visual users (va-accordion.css)
```
div.sr-button-result {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0,0,0,0);
  border: 0;
}
```

**Comment by davidakennedy** [comment]
Hi there, @powellkerry! I saw your [note about this in #accessibility-help](https://dsva.slack.com/archives/C8E985R32/p1681144579214819), and wanted to add my thoughts here as well.

I don’t think this is a bug, more of a difference in the way JAWS works vs. other screen readers. For example, VoiceOver and NDVA announce the button text change. JAWS does not. I wouldn’t expect an additional announcement from a screen reader (via aria-live) that the button did its thing because I’ve already pressed the button. It feels like we’d be getting too verbose with the output there for screen readers. I wouldn't recommend going the additional announcement route, because on other screen readers, it will get really verbose.

It wouldn't be a bad thing to do some more research here to verify that this is the way JAWS intends things to work. I did look [in its GitHub repo](https://github.com/FreedomScientific/VFO-standards-support/), and couldn't find any related bugs.

One other possible solution would be to add an `aria-expanded="true"` to the button with an `aria-controls="id1 id2"` with those IDs pointing to all the expanded/collapsed accordions that the button triggered. Not sure if that would be the best solution, but one that could work, and maybe make it not as verbose as an individually written announcement.

cc: @SarahKay8 @sara-amanda You all may want to weigh in here as well.

**Comment by caw310** [comment]
Will carry over as this was a customer support ticket that was only started at end of the sprint.
